src/main/java/org/apache/sling/cli/impl/http/HttpClientFactory.java

```diff
@@ -61,9 +67,18 @@ protected void activate(ComponentContext ctx) {
     }
 
     public CloseableHttpClient newClient() {
-        
+        final AtomicReference<String> url = new AtomicReference<>();
         return HttpClients.custom()
                 .setDefaultCredentialsProvider(newCredentialsProvider())
+                .addInterceptorFirst(
+                        (HttpRequestInterceptor) (request, context) ->
+                        url.set(((HttpRequestWrapper) request).getOriginal().getRequestLine().getUri())
+                )
+                .addInterceptorFirst((HttpResponseInterceptor) (response, context) -> {
+                    if (response.getStatusLine().getStatusCode() == HttpStatus.SC_UNAUTHORIZED) {
+                        throw new IllegalStateException("Please check your authentication details for " + url.get());
```

Comment:
Would be good to also add something like "Server returned a 401 status"

Reply:
ACK.